idb/migration/migration.go

```diff
@@ -0,0 +1,63 @@
+package migration
+
+import (
+	"fmt"
+)
+
+// A migration function should take care of writing back to metastate migration row
+type Handler func() error
+
+type Task struct {
+	MigrationId int
+
+	Handler Handler
+
+	// The system should wait for this migration to finish before trying to import new data or serve data.
+	PreventStartup bool
+
+	// Description of the migration
+	Description string
+}
+
+func RunMigrations(errChan chan <-error, statusChan chan <-string, doneChan chan <-struct{}, unblockChan chan <-struct{}, migrations ...Task) {
```

Comment:
Per discussion with @algobolson I'm going to do some more experimenting with this function. The granular channels are great for testing but make it cumbersome to use.